code/edge/src/main/java/com/adobe/marketing/mobile/EdgeNetworkService.java

```diff
@@ -577,4 +579,11 @@ private String readInputStream(final InputStream inputStream) {
 			return composeGenericErrorAsJson(e.getMessage());
 		}
 	}
+
+    private static Set<Integer> mergeUnique(final List<Integer> arr1, final List<Integer> arr2) {
```

Comment:
nit: You could make this generic, `private static <T> Set<T> mergeUnique(final List<T> arr1, final List<T> arr2) { ...`

Reply:
I am thinking it will increase testing unnecessarily, shall we refactor this later if we have a need for other types?